app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java

```diff
@@ -435,12 +435,15 @@ protected void onResume() {
         setNotificationCount();
     }
 
+
+
     @Override
     protected void onDestroy() {
         quizChecker.cleanup();
         locationManager.unregisterLocationManager();
         // Remove ourself from hashmap to prevent memory leaks
         locationManager = null;
+        NearbyParentFragmentPresenter.getInstance().removeNearbyPreferences();
```

Comment:
Do this in onStop(), onDestroy is not to be relied upon

Reply:
If I do this on onStop(), it will cause unwanted behaviour (not filling the fields when the user leaves the app after getting in the photo activity and comes back and takes the photo). Are you sure you want me to move that line ?

Reply:
AFAIF, edit text retain their states in such cases, if because of anything which we have done, they do not, we can save use savedInstanceState to retain such things

Reply:
To be honest, my tablet's broken right now so I can't really test anything new, plus I don't think I will be working more on this since I have other projects coming up. I also don't really see how you would do what you are describing. If you want me to just add a 'onStop()' function and move my line there I can do it, but that's about all I can really do, and as I said earlier, I doubt it would be a good solution.
Finally, I don't quite understand why we should not be relying on onDestroy(), especially since the code already makes use of it.